### ISSUE TITLE
Predbat time labels in the plan don't reflect DST change

### ISSUE THREAD

**Issue by gcoan** [springfall2008/batpred#1566]
**Describe the bug**
Low priority as it only happens twice a year.

The Octopus Agile rates correctly reflect the double 1:00 and 1:30 time slots as the clocks go back from BST to UTC:
![image](https://github.com/user-attachments/assets/eb404248-3abf-47c5-a0f0-482d3bc90725)
![image](https://github.com/user-attachments/assets/62f69433-ecb9-4c41-8ea0-c4bae1f07612)

The Predbat plan however labels the rates at the wrong times, the second 1:00 slot is labelled as 2:00, the 2:00 rate as 3:00 etc

![image](https://github.com/user-attachments/assets/b606f704-32f1-4dbf-8cda-fd92f06faa4e)

**Expected behavior**
Correct handling of DST changes with correct rates reflected 

I suspect it'll sort itself out after 2:00

**Predbat version**
8.5.1

**Environment details**
HAOS 2024.10.3

**Comment by springfall2008** [comment]
Yes right, the plan is showing times as minutes from midnight but ignores the time change
